torchgeo/trainers/change.py

```diff
@@ -0,0 +1,255 @@
+        pos_weight: Tensor | None = None,
+        loss: str = 'bce',
+        lr: float = 1e-3,
+        patience: int = 10,
+        freeze_backbone: bool = False,
+        freeze_decoder: bool = False,
+        threshold: float = 0.5,
+    ) -> None:
+        """Inititalize a new ChangeDetectionTask instance.
+
+        Args:
+            model: Name of the model to use.
+            backbone: Name of the `timm
+                <https://smp.readthedocs.io/en/latest/encoders_timm.html>`__ or `smp
+                <https://smp.readthedocs.io/en/latest/encoders.html>`__ backbone to use.
+            weights: Initial model weights. Either a weight enum, the string
+                representation of a weight enum, True for ImageNet weights, False or
+                None for random weights, or the path to a saved model state dict. FCN
+                model does not support pretrained weights. Pretrained ViT weight enums
+                are not supported yet.
+            in_channels: Number of input channels to model.
+            pos_weight: A weight of positive examples and used with 'bce' loss.
+            loss: Name of the loss function, currently supports
+                'bce', 'jaccard', or 'focal' loss.
+            lr: Learning rate for optimizer.
+            patience: Patience for learning rate scheduler.
+            freeze_backbone: Freeze the backbone network to fine-tune the
+                decoder and segmentation head.
+            freeze_decoder: Freeze the decoder network to linear probe
+                the segmentation head.
+            threshold: Decision threshold used to convert predicted probabilities into binary
+                class labels. Predictions greater than or equal to the threshold are assigned
+                class 1, and those below are assigned class 0. Must be between 0 and 1.
+
+        .. versionadded: 0.7
+        """
+        self.weights = weights
+        super().__init__()
+
+    def configure_losses(self) -> None:
+        """Initialize the loss criterion.
+
+        Raises:
+            ValueError: If *loss* is invalid.
+        """
+        loss: str = self.hparams['loss']
+        if loss == 'bce':
+            self.criterion = nn.BCEWithLogitsLoss(pos_weight=self.hparams['pos_weight'])
+        elif loss == 'jaccard':
+            self.criterion = smp.losses.JaccardLoss(mode='binary')
+        elif loss == 'focal':
+            self.criterion = smp.losses.FocalLoss(mode='binary', normalized=True)
+        else:
+            raise ValueError(
+                f"Loss type '{loss}' is not valid. "
+                "Currently, supports 'bce', 'jaccard', or 'focal' loss."
+            )
+
+    def configure_metrics(self) -> None:
+        """Initialize the performance metrics."""
+        metrics = MetricCollection(
+            {
+                'accuracy': BinaryAccuracy(),
+                'jaccard': BinaryJaccardIndex(),
+                'f1': BinaryF1Score(),
```

Comment:
Do we have any non-binary change detection datasets in TorchGeo? I imagine that some datasets exist with at least 3 classes (e.g., no change, deforestation, reforestation). It would be cool to support those too, identical to how we handled this in SemanticSegmentationTask which now supports binary, multiclass, and multilabel.